mlir/tools/mlir-miopen-driver/mlir-miopen-driver.cpp

```diff
@@ -290,6 +309,40 @@ static void correctParameters() {
     else
       inputLayout.setValue("g" + inputLayoutValue);
   }
+
```

Comment:
Desirably, the entire `verifyLayout()` and `correctParameters()` should stay in `Conv2dGenerator`, not as a independent function in the `mlir-miopen-driver`.

`Conv2dGenerator`, before populating the `Conv2d` op, should have every information to validate and correct parameter. This make sure that the `mlir-miopen-driver` and `mlir-miopen-lib` can take advantage of the logic similarly. 

However, it is not your fault to include this logic in here. So you can ignore this comment. Let me file a refactoring ticket to make sure `verifyLayout()`, `correctParameters()` are in the right place.